graylog2-server/src/main/java/org/graylog2/rest/resources/streams/destinations/filters/StreamDestinationFilterBuilderResource.java

```diff
@@ -0,0 +1,131 @@
+@Path("/streams/destinations/filters/builder")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@RequiresAuthentication
+public class StreamDestinationFilterBuilderResource extends RestResource {
+    // We want to control the available conditions for the stream destination filters to avoid exposing functions
+    // that don't make sense in the destination filter context.
+    // Check "/api/system/pipelines/rulebuilder/conditions" for all available conditions.
+    private static final Set<String> INCLUDED_CONDITIONS = ImmutableSet.<String>builder()
+            .add("array_contains")
+            .add("field_cidr")
+            .add("field_contains")
+            .add("field_ends_with")
+            .add("field_ip")
+            .add("field_not_null")
+            .add("field_null")
+            .add("field_starts_with")
+            .add("field_url")
+            .add("from_forwarder_input")
+            .add("from_input")
+            .add("grok_matches")
+            .add("has_field")
+            .add("has_field_equals")
+            .add("has_field_greater_or_equal")
+            .add("has_field_less_or_equal")
+            .add("lookup_has_value")
+            .add("lookup_string_list_contains")
+            .build();
+
+    private final RuleBuilderRegistry ruleBuilderRegistry;
+    private final ValidatorService validatorService;
+    private final RuleBuilderService ruleBuilderService;
+
+    @Inject
+    public StreamDestinationFilterBuilderResource(RuleBuilderRegistry ruleBuilderRegistry,
+                                                  ValidatorService validatorService,
+                                                  RuleBuilderService ruleBuilderService) {
+        this.ruleBuilderRegistry = ruleBuilderRegistry;
+        this.validatorService = validatorService;
+        this.ruleBuilderService = ruleBuilderService;
+    }
+
+    @GET
+    @Path("/conditions")
+    @ApiOperation(value = "Get available filter rule conditions")
+    @RequiresPermissions(RestPermissions.STREAM_DESTINATION_FILTERS_READ)
+    public Response getConditions() {
+        final var conditions = ruleBuilderRegistry.conditions()
+                .values()
+                .stream()
+                .map(RuleFragment::descriptor)
+                .filter(descriptor -> INCLUDED_CONDITIONS.contains(descriptor.name()))
+                .sorted(Comparator.comparing(descriptor -> requireNonNullElse(descriptor.ruleBuilderName(), descriptor.name())))
+                .toList();
+
+        return Response.ok(Map.of("conditions", conditions)).build();
+    }
+
+    @POST
+    @Path("/validate")
+    @ApiOperation("Validate rule builder")
+    @NoAuditEvent("No data changes. Only used to validate a rule builder.")
+    public Response validateRule(@ApiParam(name = "rule", required = true) @NotNull RuleBuilderDto ruleBuilderDto) {
+        final var validatedDto = validatorService.validate(ruleBuilderDto);
```

Comment:
That works. I am not sure how much work it is for Ousmane to provide different DTO's here, should we do refactoring in another ticket so that we can, also validate on saving & updating filters

Reply:
Which different DTOs do you mean? The `RuleBuilderDto` is the same type as in the regular rule builder, no?

Reply:
@kodjo-anipah Ah, because we use the `StreamDestinationFilterRuleDTO` to create the filters?

Reply:
My thought was that the `/validate` endpoint should take the standard rule builder payload. I thought that's like a built-in feature of the rule builder UI.

@ousmaneo What format would you post to the `/validate` endpoint?

Reply:
@bernd, we are posting this object to the /validate endpoint 
```
type RuleBuilderRule = {
  rule_builder: RuleBuilderType,
  description?: string,
  created_at?: string,
  id?: string,
  source?: string,
  title: string,
  modified_at?: string,
  simulator_message?: string,
}
```

Reply:
@ousmaneo Aye, that's what the endpoint is expecting. Thank you!

@kodjo-anipah That answers your question?

Reply:
@bernd  yes, but you are right sending that shouldn't be an issue on the frontend